### PR TITLE
UPD: '..' supported in Quick View

### PR DESCRIPTION
**Pull request by rich2014**
'..' supported in Quick View
1. exception raised and 'Function not supported!' displayed, when '..' in Quick View
2. show parent path now
3. tested on MacOS 12 and Windows 11

### CODE REVIEW

**Comment by rich2014** [comment]
image

**Comment by rich2014** [comment]
image

File: src/uquickviewpanel.pas [hunk 1]
```diff
@@ -166,27 +166,32 @@ procedure TQuickViewPanel.FileViewChangeActiveFile(Sender: TFileView; const aFil
...
 begin
   try
-    if not (Assigned(aFile) and aFile.IsNameValid) then
+    if not Assigned(aFile) then
       raise EAbort.Create(rsMsgErrNotSupported);
 
+    ActiveFile:= aFile.Clone;
+    if not ActiveFile.IsNameValid then
+    begin
+      ActiveFile:= TFile.Create( Sender.FileSource.GetParentDir(ActiveFile.Path) );
```
Comment (alexx2000) [inline src/uquickviewpanel.pas:175]: Memory leak

**Comment by alexx2000** [comment]
It does not work for FileSource root. You can open FTP-plugin at the panel (as example), execute QuickView and move cursor up/down, it will crash when choose `'..'` at second time.

**Comment by rich2014** [comment]
modified to only support '..' in FileSystemFileSource

File: src/uquickviewpanel.pas [hunk 2]
```diff
@@ -215,7 +215,14 @@ procedure TQuickViewPanel.FileViewChangeActiveFile(Sender: TFileView; const aFil
       else begin
         // We can use the file source directly.
         FFileSource := Sender.FileSource;
-        ActiveFile:= aFile.Clone;
+        if aFile.IsNameValid then begin
+           ActiveFile:= aFile.Clone;
+        end else begin
+          if FFileSource.IsPathAtRoot(FFileSource.GetParentDir(aFile.Path)) then
```
Comment (alexx2000) [inline src/uquickviewpanel.pas:221]: Why this special condition used? Why not show `/` root directory as is? This code works fine:
```
        if aFile.IsNameValid then begin
           ActiveFile:= aFile.Clone;
        end else begin
          ActiveFile:= TFile.Create( FFileSource.GetParentDir(aFile.Path) );
        end;
```
Am I missing something?

**Comment by alexx2000** [comment]
?

**Comment by rich2014** [comment]
any ideas ?

image

**Comment by alexx2000** [comment]
Strange that this post invisible for me. As I remember it works fine when `..` point to root directory on the Linux.
Yes, it does not work under Windows.

**Comment by rich2014** [comment]
Another PR will be submitted.

#615 closed.

**Pull request closed without merging**